pkg/operator/controller/gatewayclass/subscription_test.go

```diff
@@ -0,0 +1,416 @@
+						ClusterServiceVersionNames: []string{
+							"servicemeshoperator.v1.0.0",
+						},
+						Approval: operatorsv1alpha1.ApprovalManual,
+						Approved: true,
+					},
+				},
+			},
+			expectDelete: []client.Object{},
+		},
+		{
+			// Multiple InstallPlans exist, and the desired InstallPlan is not approved. Expect InstallPlan.Spec.Approved = true only for the desired InstallPlan
+			name:    "Multiple InstallPlans",
+			channel: "stable",
+			version: "servicemeshoperator.v1.0.0",
+			existingObjects: []runtime.Object{
+				&operatorsv1alpha1.Subscription{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: operatorcontroller.ServiceMeshSubscriptionName().Namespace,
+						Name:      operatorcontroller.ServiceMeshSubscriptionName().Name,
+						UID:       "foobar",
+					},
+					Spec: &operatorsv1alpha1.SubscriptionSpec{
+						Channel:                "stable",
+						InstallPlanApproval:    operatorsv1alpha1.ApprovalManual,
+						Package:                "servicemeshoperator",
+						CatalogSource:          "redhat-operators",
+						CatalogSourceNamespace: "openshift-marketplace",
+						StartingCSV:            "servicemeshoperator.v1.0.0",
+					},
+				},
+				&operatorsv1alpha1.InstallPlan{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "Install-foo",
+						Namespace: operatorcontroller.OpenshiftOperatorNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								UID: "foobar",
+							},
+						},
+					},
+					Spec: operatorsv1alpha1.InstallPlanSpec{
+						ClusterServiceVersionNames: []string{
+							"servicemeshoperator.v1.0.0",
+						},
+						Approval: operatorsv1alpha1.ApprovalManual,
+						Approved: false,
+					},
+				},
+				&operatorsv1alpha1.InstallPlan{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "Install-bar",
+						Namespace: operatorcontroller.OpenshiftOperatorNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								UID: "foobar",
+							},
+						},
+					},
+					Spec: operatorsv1alpha1.InstallPlanSpec{
+						ClusterServiceVersionNames: []string{
+							"servicemeshoperator.v1.1.0",
+						},
+						Approval: operatorsv1alpha1.ApprovalManual,
+						Approved: false,
+					},
+				},
+			},
+			expectCreate: []client.Object{},
+			expectUpdate: []client.Object{
+				&operatorsv1alpha1.InstallPlan{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "Install-foo",
+						Namespace: operatorcontroller.OpenshiftOperatorNamespace,
+						OwnerReferences: []metav1.OwnerReference{
+							{
+								UID: "foobar",
+							},
+						},
+					},
+					Spec: operatorsv1alpha1.InstallPlanSpec{
+						ClusterServiceVersionNames: []string{
+							"servicemeshoperator.v1.0.0",
+						},
+						Approval: operatorsv1alpha1.ApprovalManual,
+						Approved: true,
+					},
+				},
+			},
+			expectDelete: []client.Object{},
+		},
+		{
+			// An InstallPlan with the correct version exists, but it's owned by a different subscription. Expect no change.
```

Comment:
What does "owned by a different subscription" mean? I see the UIDs are different, are the UIDs are good indicator of the owership?

Reply:
The main concern with "owned by a different subscription" is if a user installed OSSM separately, but in theory there could also be another OLM package that includes OSSM in its install plan. When OLM creates install plans, it includes an owner reference to the subscription that it was created for.

If I'm reading the k8s docs correctly (for completeness, [short description of UIDs here](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#uids)), UIDs for a given resource are guaranteed to be unique, so matching the subscription's UID to the UID in an owner reference is sufficient to make sure we only approve install plans for the subscription created by the ingress operator.